src/cli/api/local_api.py

```diff
@@ -382,20 +526,266 @@ def get_results(self, path: Path, all_results: bool) -> List[ResultType]:
+                                         f"{len(network_nodes)}")
+
+        # Check if all nodes exist and belong to this network
+        invalid_nodes = []
+        for node in experiment_nodes:
+            if node['slug'] not in network_nodes:
+                invalid_nodes.append(node['slug'])
+        if invalid_nodes:
+            if len(invalid_nodes) > 1:
+                error_dict['warning'].append(
+                    f"In file {experiment_file_path}: {invalid_nodes} do not exist or do not belong to"
+                    f" the network '{experiment_network_slug}'")
+            elif len(invalid_nodes) == 1:
+                error_dict['warning'].append(f"In file {experiment_file_path}: {invalid_nodes} does not exist or does "
+                                             f"not belong to the network '{experiment_network_slug}'")
+
+    def validate_experiment_channels(self, experiment_file_path: Path, experiment_data: Dict[str, Any], error_dict:
+                                     ErrorDictType) -> None:
+        """
+        Validate if the amount of channels (defined in the experiment.json file) are valid for 'network_slug' and if
+        all the channels exist and belong to this network.
+
+        Args:
+            experiment_file_path: The location of the experiment.json file
+            experiment_data: contents of the experiment.json file
+            error_dict: Dictionary containing error and warning messages of the validations that failed
+        """
+
+        # Check if the amount of channels are valid for this network
+        experiment_network_slug = experiment_data['asset']['network']['slug']
+        experiment_channels = experiment_data['asset']['network']['channels']
+        network_channels = self._get_channels_for_network(experiment_network_slug)
+
+        if len(experiment_channels) > len(network_channels):
+            error_dict['warning'].append(f"In file {experiment_file_path}: too many channels used in network "
+                                         f"'{experiment_network_slug}'. Maximum amount of channels that can be used: "
+                                         f"{len(network_channels)}")
+
+        # Check if the channels exist and belong to this network
+        invalid_channels = []
+        for channel in experiment_channels:
+            # Replace '_' with '-'. Eg: the_hague can be written this way in experiment.json while the comparison is
+            # done with the-hague defined in network/channels.json
+            node_slug1 = channel['node_slug1'].replace('_', '-')
+            node_slug2 = channel['node_slug2'].replace('_', '-')
+            if not node_slug1 + '-' + node_slug2 in network_channels:
+                if not node_slug2 + '-' + node_slug1 in network_channels:
+                    invalid_channels.append([channel['node_slug1'], channel['node_slug2']])
+
+        if invalid_channels:
```

Comment:
Same here: Instead of this check for plural form etc. I opt for adding a line to the error_dict for each wrong channel. You can already do that in line 750. You don't need the invalid_channels list.  Also printing a list you get the [], which we would not have when printing each message separate.